pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/SortOperatorTest.java

```diff
@@ -0,0 +1,363 @@
+
+    Mockito.when(_input.nextBlock())
+        .thenReturn(block(schema, new Object[]{2}, new Object[]{1}, new Object[]{3}))
+        .thenReturn(TransferableBlockUtils.getEndOfStreamTransferableBlock());
+
+    // When:
+    op.nextBlock(); // consume, create NOOP
+    TransferableBlock block = op.nextBlock(); // construct
+    TransferableBlock block2 = op.nextBlock(); // eos
+
+    // Then:
+    Assert.assertEquals(block.getNumRows(), 3);
+    Assert.assertTrue(block2.isEndOfStreamBlock(), "expected EOS block to propagate");
+  }
+
+  @Test
+  public void shouldConsumeAndSortTwoInputBlocksWithOneRowEach() {
+    // Given:
+    List<RexExpression> collation = collation(0);
+    List<Direction> directions = ImmutableList.of(Direction.ASCENDING);
+    DataSchema schema = new DataSchema(new String[]{"sort"}, new DataSchema.ColumnDataType[]{INT});
+    SortOperator op = new SortOperator(_input, collation, directions, 10, 0, schema);
+
+    Mockito.when(_input.nextBlock())
+        .thenReturn(block(schema, new Object[]{2}))
+        .thenReturn(block(schema, new Object[]{1}))
+        .thenReturn(TransferableBlockUtils.getEndOfStreamTransferableBlock());
+
+    // When:
+    op.nextBlock(); // consume, create NOOP
+    op.nextBlock(); // consume, create NOOP
+    TransferableBlock block = op.nextBlock(); // construct
+    TransferableBlock block2 = op.nextBlock(); // eos
+
+    // Then:
+    Assert.assertEquals(block.getNumRows(), 2);
+    Assert.assertEquals(block.getContainer().get(0), new Object[]{1});
+    Assert.assertEquals(block.getContainer().get(1), new Object[]{2});
+    Assert.assertTrue(block2.isEndOfStreamBlock(), "expected EOS block to propagate");
+  }
+
+  @Test
+  public void shouldBreakTiesUsingSecondCollationKey() {
+    // Given:
+    List<RexExpression> collation = collation(0);
+    List<Direction> directions = ImmutableList.of(Direction.ASCENDING);
+    DataSchema schema = new DataSchema(new String[]{"first", "second"}, new DataSchema.ColumnDataType[]{INT, INT});
+    SortOperator op = new SortOperator(_input, collation, directions, 10, 0, schema);
+
+    Mockito.when(_input.nextBlock())
+        .thenReturn(block(schema, new Object[]{1, 2}, new Object[]{1, 1}))
+        .thenReturn(TransferableBlockUtils.getEndOfStreamTransferableBlock());
+
+    // When:
+    op.nextBlock(); // consume, create NOOP
+    TransferableBlock block = op.nextBlock(); // construct
+    TransferableBlock block2 = op.nextBlock(); // eos
+
+    // Then:
+    Assert.assertEquals(block.getNumRows(), 2);
+    Assert.assertEquals(block.getContainer().get(0), new Object[]{1, 1});
+    Assert.assertEquals(block.getContainer().get(1), new Object[]{1, 2});
+    Assert.assertTrue(block2.isEndOfStreamBlock(), "expected EOS block to propagate");
+  }
+
+  @Test
+  public void shouldHandleNoOpUpstreamBlockWhileConstructing() {
+    // Given:
+    List<RexExpression> collation = collation(0);
```

Comment:
🤦 I added a test for this (`shouldBreakTiesUsingSecondCollationKey`) but I didn't add the second collation key at all! It passed because of sheer luck haha. Fixed and added a mixed ASCENDING/DESCENDING test